airflow/providers/ssh/hooks/ssh.py

```diff
@@ -275,3 +281,36 @@ def create_tunnel(
                       category=DeprecationWarning)
 
         return self.get_tunnel(remote_port, remote_host, local_port)
+
+    @staticmethod
+    def _add_new_record_to_known_hosts(record, file):
+        file.write(''.join([record, '\n']))
+
+    @staticmethod
+    def add_host_to_known_hosts(host, key_type, host_key):
+        """This adds a specified remote_host public key to the known_hosts
+            in order to prevent man-in-the-middle attacks."""
+        # The .ssh hidden directory is required and not present on all airflow deployments
+        known_hosts_file_ref = SSHHook._create_known_hosts()
```

Comment:
Is there assumption here that this method always successful?
Can there be cases where it throws exception that isn't handled?

Reply:
@RosterIn Thanks for the input. Yes, I was assuming the user would have write access to everything under their home dir but there are certainly instances where this may not be the case which I hadn't considered such as default umask for example. Changes made.

Reply:
Aplogies, i forced pushed a rebase and the commit links are now dead - https://github.com/apache/airflow/pull/7739/commits/fc973414d5e2ca77c9dd927d867286c73e320629#diff-5b40f2980778e7de291e925b11fb14e7R294